scripts/levelManagement.js

```diff
@@ -0,0 +1,136 @@
+        event.preventDefault();
+    });
+}
+
+function getSelectedLevel(responseJSON) {
+    var level = JSON.parse(responseJSON)["level-name"];
+    document.getElementById("selected-level").innerHTML = level;
+    configureDownloadLevelButton();
+    configureDeleteLevelButton();
+}
+
+function getAvailableLevels(responseJSON) {
+    var levels = JSON.parse(responseJSON)["levels"];
+    document.getElementById("available-levels").innerHTML = "";
+    for (level of levels) {
+        var level_name = level["level-name"];
+        var buttonId = `level-name=${level_name}`;
+        document.getElementById("available-levels").innerHTML += newButton(
+            buttonId,
+            level_name
+        );
+    }
+    for (level of levels) {
+        var level_name = level["level-name"];
+        var buttonId = `level-name=${level_name}`;
+        var button = document.getElementById(buttonId);
+        button.addEventListener("click", (e) => {
+            var url = `/cgi-bin/setLevel.py?${e["target"]["id"]}`;
+            var request = new XMLHttpRequest();
+            request.open("GET", url, true);
+            request.onload = function () {
+                var res = JSON.parse(request.responseText);
+                if (res["success"] === "true") {
+                    updateLevelModule();
+                }
+            };
+            request.onerror = function () {};
+            request.send(); // create FormData from form that triggered event
+            event.preventDefault();
+        });
+    }
+}
+
+function setLevelName(event) {
+    var url = "/cgi-bin/setLevel.py";
+    var request = new XMLHttpRequest();
+    request.open("POST", url, true);
+    request.onload = function () {
+        // request successful
+        // we can use server response to our request now
+        updateLevelModule();
+    };
+
+    request.onerror = function () {
+        // request failed
+    };
+
+    request.send(new FormData(event.target)); // create FormData from form that triggered event
+    event.preventDefault();
+}
+
+function uploadLevel(event) {
+    var url = "/cgi-bin/uploadLevel.py";
+    var request = new XMLHttpRequest();
+    request.open("POST", url, true);
+    request.onload = function () {
+        var res = JSON.parse(request.responseText);
+        if (res["success"] === "true") {
+            document.getElementById(
+                "upload-level-success"
+            ).innerText = `Upload successful. (${getCurrentDatetime()})`;
+            updateLevelModule();
+        }
+    };
+
+    request.onerror = function () {
+        // request failed
+    };
+
+    request.send(new FormData(event.target)); // create FormData from form that triggered event
+    event.preventDefault();
+}
+
+function updateLevelModule() {
+    httpGetAsync("/cgi-bin/getCurrentLevel.py", getSelectedLevel);
+    httpGetAsync("/cgi-bin/getLevelsList.py", getAvailableLevels);
```

Comment:
Another possible name for the functions mentioned above are ```updateSelectedLevel``` and ```updateAvailableLevels```

Reply:
I agree.